dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java

```diff
@@ -124,70 +156,98 @@ public static DependencyGraph getCompleteDependencies(Artifact artifact)
    */
   public static DependencyGraph getTransitiveDependencies(Artifact artifact)
       throws DependencyCollectionException, DependencyResolutionException {
-    
     // root node
-    DependencyNode node = resolveCompileTimeDependencies(artifact);  
+    DependencyNode node = resolveCompileTimeRootDependencies(artifact);
     DependencyGraph graph = new DependencyGraph();
-    preorder(new Stack<>(), node, graph);
-    
+    levelorder(node, graph);
     return graph;
   }
-  
-  // TODO Dedup the next two methods. They are duplicate code with only one line difference.
-  // this finds the actual graph that Maven sees with no duplicates and at most one version per
-  // library.
-  @SuppressWarnings("unchecked")
-  private static void preorder(Stack<DependencyNode> path, DependencyNode current,
-      DependencyGraph graph) {
-    
-    path.push(current);
-    DependencyPath forPath = new DependencyPath();
-    for (DependencyNode node : path) {
-      if (!"system".equals(node.getDependency().getScope())) {
-        forPath.add(node.getArtifact());
-      }
+
+  static DependencyGraph getTransitiveDependencies(List<Artifact> artifacts)
+      throws DependencyCollectionException, DependencyResolutionException {
+    // root node is dummy (artifact: null) and has dependencies as children
+    DependencyNode node = resolveCompileTimeDependencies(artifacts);
+    DependencyGraph graph = new DependencyGraph();
+    levelorder(node, graph);
+    return graph;
+  }
+
+  private static class LevelOrderQueueItem {
+    DependencyNode dependencyNode;
```

Comment:
final

Reply:
done

Reply:
Oh, I meant make the fields final. You finalized the class (which is fine).